### DISCUSSION TITLE
Roadmap for arquillian-warp

### DISCUSSION THREAD

**Discussion by WolfgangHG** [arquillian/arquillian-extension-warp#115] [Ideas]
@chengfang I start this discussion thread to coordinate about the roadmap for the extension as a followup of your comment on pull request #114 

I started working on this extension with the goal to fix https://issues.redhat.com/browse/ARQ-2216 - this should be only a "simple" dependency update. But then I got distracted and am still not finished with maintenance work ;-).
My plan would to fix the Javassist issue and let someone make a minor release which repairs compatibility with Java 11 and beyond.

The next big step for version 2.0 will be an update to JakartaEE 9 or 10 and a switch to Java 11.

What do you think?

**Comment by chengfang** [comment]
@WolfgangHG thanks for bringing that up.

It currently supports JDK 8. I think we should be able to cut a minor release with all the fixes we've had so far.

Javassist fix and JDK 11 support will be included in the next minor release.

Support for EE 9, 10 can be next major releases.

I'll try to put them into a github `Project`. If anyone has any additional feature requests or bug fixes, feel free to create issues so that they can be planned accordingly.

**Reply by rhusar** [reply]
Reviewing the changes now, it sounds like a micro to me - https://github.com/arquillian/arquillian-extension-warp/compare/1.0.0...main - will cut 1.0.1.Final - https://github.com/arquillian/arquillian-extension-warp/issues/194

**Comment by WolfgangHG** [comment]
After switching to JakartaEE9 und Java 11: The "animal-sniffer-maven-plugin" is probably no longer relevant and can be removed, as the [project page](https://github.com/mojohaus/animal-sniffer) states:

_Please note this plugin is now in maintenance level as the exact same feature can be now easily achieved using the --release flag from Javac_

There are also no Java signatures newer than 1.8. available.

**Comment by chengfang** [comment]
Looks like we're ready for 1.1.0 release (current release version is 1.0.0). I think we've made some great progress to warrant a minor version bump. What do you all think?

**Reply by WolfgangHG** [reply]
I would like to sort out the LittleProxy issue - but still without success.

And the final TODO (javassist update) is still open, though I hope that this is trivial ;-).

**Reply by WolfgangHG** [reply]
I think I will delay further research for the LittleProxy problem - it is rather frustrating currently to make no progress....

Remaining (optional) TODOs: 
-37 Java warnings by Eclipse
-I did not update the GlassFish profiles to version 5, because each test took 2 minutes to run. GlassFish 4 does not show this problem. I would do some research whether this can be improved.

**Comment by WolfgangHG** [comment]
Just to give a bit of feedback: currently I don't have time to work on this extension - some renovation work at home for the next weeks ;-).

The LittleProxy (#134) issue is rather frustrating - as I don't manage to build Netty myself, I cannot add debugging code to track the problem down. But this should not block a release.

And before doing the step to JakartaEE 10, I would like to see the #131 fixed, because currently we have a working test for GlassFish 4 but cannot upgrade to newer versions without a major performance impact on the tests for GlassFish. If we move to JakartaEE10, we would have to switch to a newer GlassFish, and thus the tests would run for about an hour.
I still hope that you can pull in someone from the Arquillian team who could help resolving this. I don't have any idea where to look.

**Comment by chengfang** [comment]
@WolfgangHG thanks for the update.

**Comment by WolfgangHG** [comment]
With the latest "warnings" and "hamcrest" pull request , I am done with all changes I think ;-).

But it would be good to wait for a release of arquillian-drone. Current update to 3.0.0.alpha7 as part of #100 was a bit too early, as firefox and chrome webdriver support is broken. We could also revert to the latest 2.x release before doing a release. @chengfang What do you think

It would also be great if we could get my two arquillian pull requests merged, but there was no comment on them up to now.

**Comment by WolfgangHG** [comment]
@chengfang Drone 3.0.0-alpha.8 is out, but it requires Java 11. So, currently support for firefox or chrome browser is broken and we cannot fix it.

My initial intention was to do a Java 1.8 release of this extension with all the changes that bring it back to life, then switch to Java 11 and JakartaEE10. 
I still wait for a review of my ThreadLocal pull request in arquillian-core ([arquillian-core#501](https://github.com/arquillian/arquillian-core/pull/501)), but there was no update since a long time. And it is not a prerequisite for a release.

So, could you or someone else do a 1.8 release?
